File: 2-Ad-Hoc/1091-Divisao-da-Nlogonia/1091.cpp
#include <iostream>

int main() {
    int cases, dx, dy, hx, hy;
    while (1) {
        std::cin>>cases;
        if (cases == 0) break;
        std::cin>>dx>>dy;
        for (int i = 0; i < cases; ++i) {
            std::cin>>hx>>hy;
            if (hx == dx || hy == dy) {
                std::cout<<"divisa\n";
            }
            else {
                if (hx > dx) {
                    if (hy > dy) {
                        std::cout<<"NE\n";
                    }
                    else {
                        std::cout<<"SE\n";
                    }
                }
                else {
                    if (hy > dy) {
                        std::cout<<"NO\n";
                    }
                    else {
                        std::cout<<"SO\n";
                    }
                }   
            }
        }
    }
    return 0;
}

File: 2-Ad-Hoc/1091-Divisao-da-Nlogonia/1091.java
import java.util.Scanner;

public class Main {
    public static void main (String[] args) {
        int cases, dx, dy, hx, hy;
        Scanner scan = new Scanner(System.in);
        while (true) {
            cases = scan.nextInt();
            if (cases == 0) break;
            dx = scan.nextInt();
            dy = scan.nextInt();
            for (int i = 0; i < cases; ++i) {
                hx = scan.nextInt();
                hy = scan.nextInt();
                if (hx == dx || hy == dy) {
                    System.out.println("divisa");
                }
                else {
                    if (hx > dx) {
                        if (hy > dy) {
                            System.out.println("NE");
                        }
                        else {
                            System.out.println("SE");
                        }
                    }
                    else {
                        if (hy > dy) {
                            System.out.println("NO");
                        }
                        else {
                            System.out.println("SO");
                        }
                    }
                }
            }
        }
    }
}